doc/Media2.xml

```diff
@@ -2167,6 +2167,244 @@
+          <listitem>
+            <para>Token [tt:ReferenceToken] – Unique Token associated with the audio clip configuration.</para>
+          </listitem>
+          <listitem>
+            <para>Type [xs:string] - Format of the audio clip. For definitions see tt:SupportedAudioClipFormat parameter.</para>
+          </listitem>          
+          <listitem>
+            <para>RepeatCycles [xs:int] - Number of times the audio clip can be repeated when it is played. The value 0 means no repeat
+            , -1 means repeat infinite i.e repeat the audio clip until it is stopped by user operation.</para>
+          </listitem>
+		  <listitem>
+            <para>Gain [xs:int] [optional] - Optional Gain to control volume of the audio clip when it is played.</para>
+          </listitem>
+		  <listitem>
+            <para>ScheduleToken [pt:ReferenceToken] [optional] - Optional Schedule token associated with the audio clip configuration. Audio clip can be played when configured schedule is triggered.</para>
+          </listitem>
+        </itemizedlist>
+        <para>The device should accept optional ScheduleToken from the client if it supports ONVIF Schedule service specification.</para>
+      </section>
+      <section xml:id="section_wys_czg_ryc">
+        <title>GetAudioClipConfigurations</title>
+        <para>This operation gets audio clip configurations from the device.		
+		If audio clip token is provided, the device shall return the audio clip configuration associated with the token.
+		If no token is provided, the device shall return all audio clip configurations exist in the device.</para>
+        <variablelist role="op">
+          <varlistentry>
+            <term>request</term>
+            <listitem>
+              <para role="param">Token - optional
+                [tt:ReferenceToken]</para>
+              <para role="text">Token associated with the audio clip configuration.</para>
+            </listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>response</term>
+            <listitem>
+              <para role="param">AudioClipsConfiguration - optional, unbounded
+                [tr2:AudioClipConfiguration]</para>
+              <para role="text">This message contains audio clip configurations of the device.</para>
+            </listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>faults</term>
+            <listitem>
+              <para role="param">env:Sender - ter:InvalidArgVal - ter:NoAudioClip</para>
+              <para role="text">The audio clip token does not exist.</para>
+            </listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>access class</term>
+            <listitem>
+              <para role="access">READ_SYSTEM</para>
+            </listitem>
+          </varlistentry>
+        </variablelist>
+      </section>
+      <section xml:id="section_wvd_dzg_ryd">
+        <title>AddAudioClipConfiguration</title>
```

Comment:
As already mentioned the interface with timeout for upload is error prone. Prefer to define static upload URI provided by capabilities. Upload sets mime-type of clip. Other parameters can be changed afterwords via SetAudioClip.

Reply:
currently upload is error prone due to two step process i.e add audio clip configuration in first step and second is uploading actual audio clip file.  Providing static upload uri in the capabilities just makes client to aware upfront where to upload the audio clip, but still the associated audio clip configuration parameters i.e Enable, Name, Type, Gain, schedule, repeat cycles need to be configured in the device in second step. the error prone problem can be solved only when single step process for both audio clip configuration and actual audio clip file. Yes, we will discuss the pros and cons for the approaches in next week VEWG telco.

Reply:
I agree that a dynamic URI should be avoided, also a description of the exact format of clip upload is missing I guess?
It would be possible to either have a plain file-upload with query parameters to pass ClipConfiguration info, or doing a request with XML payload with the filedata encoded using hexBinary or base64, or have a multipart upload with one XML part and one file data part
Is this the first API call with file upload within the ONVIF specs?
In either case having the ability to have a dedicated URI for the upload is probably a good thing.

Reply:
StartFirmwareUpgrade response  includes an HTTP URL to which the upgrade file may be uploaded.
https://www.onvif.org/ver10/device/wsdl/devicemgmt.wsdl. 

Carrying actual audio clip file content as base 64 inside soap/xml will definitely have compatibility issues due to size of soap/xml devices has to handle.  this approach we are aware in the first day of spec definition.